diff --git a/third_party/msquic/src/platform/cert_capi.c b/third_party/msquic/src/platform/cert_capi.c
index 184035e81..d5d2e821a 100644
--- a/third_party/msquic/src/platform/cert_capi.c
+++ b/third_party/msquic/src/platform/cert_capi.c
@@ -162,6 +162,7 @@ HashAlgFromTLS(
     }
 }
 
+#if (NTDDI_VERSION >= NTDDI_WIN8)
 static
 _Success_(return != NULL)
 BCRYPT_ALG_HANDLE
@@ -197,6 +198,7 @@ HashHandleFromTLS(
         return NULL;
     }
 }
+#endif // NTDDI_VERSION >= NTDDI_WIN8
 
 #define CXPLAT_CERTIFICATE_MAX_HASH_SIZE 64
 
@@ -1429,6 +1431,19 @@ CxPlatCertVerify(
     _In_ size_t SignatureLength
     )
 {
+#if (NTDDI_VERSION < NTDDI_WIN8)
+    UNREFERENCED_PARAMETER(Certificate);
+    UNREFERENCED_PARAMETER(SignatureAlgorithm);
+    UNREFERENCED_PARAMETER(CertListToVerify);
+    UNREFERENCED_PARAMETER(CertListToVerifyLength);
+    UNREFERENCED_PARAMETER(Signature);
+    UNREFERENCED_PARAMETER(SignatureLength);
+    QuicTraceEvent(
+        LibraryError,
+        "[ lib] ERROR, %s.",
+        "CxPlatCertVerify requires Windows 8 or later");
+    return FALSE;
+#else
     PCCERT_CONTEXT CertCtx = (PCCERT_CONTEXT)Certificate;
 
     QuicTraceLogVerbose(
@@ -1561,4 +1576,5 @@ Exit:
     }
 
     return Result;
+#endif // NTDDI_VERSION >= NTDDI_WIN8
 }
